File: puzzles/day 1/python/one.py

elfCallorydb = []
CalloryPerElfdb = []
with open("../input.txt", 'r') as f:
    input = f.readlines()
    elfCallorydb = input

totalPerElf = 0
for value in elfCallorydb:
    if value != '\n':
        totalPerElf += int(value.replace('\n', ''))
    else:
        CalloryPerElfdb.append(totalPerElf)
        totalPerElf = 0
CalloryPerElfdb.sort(reverse=True)
print(CalloryPerElfdb[0])

File: puzzles/day 1/one.js
const fs = require("fs");
var elfCallorydb = []

fs.readFile('./input.txt', 'utf8',(err,data)=>{
    if(err)return console.log(err);
    const arrayOfInput = data.split("\n")
    let totalPerElf = 0;
    arrayOfInput.forEach(calloryItem => {
    if(calloryItem > 0){
        totalPerElf+=parseInt(calloryItem)
    }else {
    elfCallorydb.push(totalPerElf)
    totalPerElf=0;
    }
    });
    let max = elfCallorydb.sort((a,b)=>a < b ? 1 : -1)[0]
    console.log(max)
})
